jetstream/analysis.py

```diff
@@ -457,3 +447,38 @@ def run(self, current_date: datetime, dry_run: bool = False) -> None:
 
         result_futures = client.compute(results)
         client.gather(result_futures)  # block until futures have finished
+
+    def ensure_enrollments(self, current_date: datetime) -> None:
+        """Ensure that enrollment tables for experiment are up-to-date or re-create."""
+        time_limits = self._get_timelimits_if_ready(AnalysisPeriod.DAY, current_date)
+
+        if time_limits is None:
+            logger.info("Skipping %s (%s); not ready", self.config.experiment.normandy_slug)
+            return
+
+        if self.config.experiment.start_date is None:
+            raise errors.NoStartDateException(self.config.experiment.normandy_slug)
+
+        normalized_slug = bq_normalize_name(self.config.experiment.normandy_slug)
+        enrollments_table = f"enrollments_{normalized_slug}"
+
+        logger.info(f"Create {enrollments_table}")
+        exp = mozanalysis.experiment.Experiment(
+            experiment_slug=self.config.experiment.normandy_slug,
+            start_date=self.config.experiment.start_date.strftime("%Y-%m-%d"),
+        )
+        enrollments_sql = exp.build_enrollments_query(
+            time_limits,
+            self.config.experiment.platform.enrollments_query_type,
+            self.config.experiment.enrollment_query,
+            self.config.experiment.segments,
+        )
+
+        try:
+            self.bigquery.execute(
+                enrollments_sql,
+                enrollments_table,
+                google.cloud.bigquery.job.WriteDisposition.WRITE_EMPTY,
+            )
+        except Conflict:
+            pass
```

Comment:
How do you feel about adding this optional `write_disposition` parameter to `bigquery.execute`?

Reply:
Yeah that seems good. I guess there's a design principle where using the Google API enums doesn't seem very well-encapsulated, but we do already expect this function to throw the Google API exceptions, so we may as well leak that abstraction consistently :D